diff -u -r patched/ContactCardPage.qml original/ContactCardPage.qml
--- patched/ContactCardPage.qml	2014-03-18 09:14:26.332500695 +0100
+++ original/ContactCardPage.qml	2014-03-17 18:21:24.193010382 +0100
@@ -3,6 +3,7 @@
 import org.nemomobile.contacts 1.0
 import Sailfish.TransferEngine 1.0
 import "contactcard/ContactsDBusService.js" as ContactsService
+import org.SfietKonstantin.friends.integration 1.0
 
 Page {
     id: root
@@ -26,6 +27,29 @@
     signal editRequested()
     signal deleteRequested()
 
+    NemoContactBridge {
+        id: contactBridge
+        property bool attachedPushed: false
+        function pushAttached() {
+            if (facebookId != "" && !attachedPushed && root.status == PageStatus.Active) {
+                var page = pageStack.pushAttached(friendsUserAttachedPage)
+                page.load()
+                attachedPushed = true
+            }
+        }
+
+        contactId: contact.id
+        onContactIdChanged: fetch()
+        onFacebookIdChanged: pushAttached()
+    }
+
+    Component {
+        id: friendsUserAttachedPage
+        FriendsUserAttachedPage {
+            facebookId: contactBridge.facebookId
+        }
+    }
+
     function showEditor() {
         ContactsService.editContact(contactId)
     }
@@ -107,6 +131,7 @@
         if (status === PageStatus.Activating && contact && contact.complete) {
             _updateContactInfo()
         }
+        contactBridge.pushAttached()
     }
 
     Component {
